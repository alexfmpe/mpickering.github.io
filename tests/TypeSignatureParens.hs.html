{-# LANGUAGE ScopedTypeVariables #-}

pTokenCost :: forall loc state a .((Show a, Eq a,  loc `IsLocationUpdatedBy` a, LL.ListLike state a) => [a] -> Int -> P (Str  a state loc) [a])
pTokenCost as cost = 5
