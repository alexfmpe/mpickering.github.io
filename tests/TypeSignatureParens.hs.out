{-# LANGUAGE ScopedTypeVariables #-}

pTokenCost :: forall loc state a . (Show a, Eq a,  loc `IsLocationUpdatedBy` a, LL.ListLike state a) => [a] -> Int -> P (Str  a state loc) [a]
pTokenCost as cost = 5

==============
tests/examples/TypeSignatureParens.hs
==============
lengths:(204,205)

==============

({ tests/examples/TypeSignatureParens.hs:1:1 }
 Just (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnEofPos),DP (1,0))])
 (HsModule 
  (Nothing) 
  (Nothing) 
  [] 
  [
   ({ tests/examples/TypeSignatureParens.hs:3:1-143 }
    Just (Ann (DP (2,1)) (ColDelta 1) DP (2,1) [DComment (DP (0,0),DP (0,36)) "{-# LANGUAGE ScopedTypeVariables #-}"] [((G AnnDcolon),DP (0,1))])
    (SigD 
     (TypeSig 
      [
       ({ tests/examples/TypeSignatureParens.hs:3:1-10 }
        Just (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnVal),DP (0,0))])
        (Unqual {OccName: pTokenCost}))] 
      ({ tests/examples/TypeSignatureParens.hs:3:15-143 }
       Just (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnForall),DP (0,0)),((G AnnDot),DP (0,1))])
       (HsForAllTy 
        (Explicit) 
        (Nothing) 
        (HsQTvs 
         [] 
         [
          ({ tests/examples/TypeSignatureParens.hs:3:22-24 }
           Just (Ann (DP (0,1)) (ColDelta 22) DP (0,1) [] [((G AnnVal),DP (0,0))])
           (UserTyVar 
            (Unqual {OccName: loc}))),
          ({ tests/examples/TypeSignatureParens.hs:3:26-30 }
           Just (Ann (DP (0,1)) (ColDelta 26) DP (0,1) [] [((G AnnVal),DP (0,0))])
           (UserTyVar 
            (Unqual {OccName: state}))),
          ({ tests/examples/TypeSignatureParens.hs:3:32 }
           Just (Ann (DP (0,1)) (ColDelta 32) DP (0,1) [] [((G AnnVal),DP (0,0))])
           (UserTyVar 
            (Unqual {OccName: a})))]) 
        ({ tests/examples/TypeSignatureParens.hs:3:36-100 }
         Just (Ann (DP (0,1)) (ColDelta 36) DP (0,1) [] [((G AnnOpenP),DP (0,0)),((G AnnCloseP),DP (0,0)),((G AnnDarrow),DP (0,1))])
         [
          ({ tests/examples/TypeSignatureParens.hs:3:37-42 }
           Just (Ann (DP (0,0)) (ColDelta 37) DP (0,0) [] [((G AnnComma),DP (0,0))])
           (HsAppTy 
            ({ tests/examples/TypeSignatureParens.hs:3:37-40 }
             Just (Ann (DP (0,0)) (ColDelta 37) DP (0,0) [] [((G AnnVal),DP (0,0))])
             (HsTyVar 
              (Unqual {OccName: Show}))) 
            ({ tests/examples/TypeSignatureParens.hs:3:42 }
             Just (Ann (DP (0,1)) (ColDelta 42) DP (0,1) [] [((G AnnVal),DP (0,0))])
             (HsTyVar 
              (Unqual {OccName: a}))))),
          ({ tests/examples/TypeSignatureParens.hs:3:45-48 }
           Just (Ann (DP (0,1)) (ColDelta 45) DP (0,1) [] [((G AnnComma),DP (0,0))])
           (HsAppTy 
            ({ tests/examples/TypeSignatureParens.hs:3:45-46 }
             Just (Ann (DP (0,0)) (ColDelta 45) DP (0,0) [] [((G AnnVal),DP (0,0))])
             (HsTyVar 
              (Unqual {OccName: Eq}))) 
            ({ tests/examples/TypeSignatureParens.hs:3:48 }
             Just (Ann (DP (0,1)) (ColDelta 48) DP (0,1) [] [((G AnnVal),DP (0,0))])
             (HsTyVar 
              (Unqual {OccName: a}))))),
          ({ tests/examples/TypeSignatureParens.hs:3:52-78 }
           Just (Ann (DP (0,2)) (ColDelta 52) DP (0,2) [] [((G AnnComma),DP (0,0))])
           (HsOpTy 
            ({ tests/examples/TypeSignatureParens.hs:3:52-54 }
             Just (Ann (DP (0,0)) (ColDelta 52) DP (0,0) [] [((G AnnVal),DP (0,0))])
             (HsTyVar 
              (Unqual {OccName: loc}))) 
            ((,) 
             (WpKiApps 
              []) 
             ({ tests/examples/TypeSignatureParens.hs:3:56-76 }
              Just (Ann (DP (0,1)) (ColDelta 56) DP (0,1) [] [((G AnnBackquote),DP (0,0)),((G AnnVal),DP (0,0)),((G AnnBackquote),DP (0,0))])
              (Unqual {OccName: IsLocationUpdatedBy}))) 
            ({ tests/examples/TypeSignatureParens.hs:3:78 }
             Just (Ann (DP (0,1)) (ColDelta 78) DP (0,1) [] [((G AnnVal),DP (0,0))])
             (HsTyVar 
              (Unqual {OccName: a}))))),
          ({ tests/examples/TypeSignatureParens.hs:3:81-99 }
           Just (Ann (DP (0,1)) (ColDelta 81) DP (0,1) [] [])
           (HsAppTy 
            ({ tests/examples/TypeSignatureParens.hs:3:81-97 }
             Just (Ann (DP (0,0)) (ColDelta 81) DP (0,0) [] [])
             (HsAppTy 
              ({ tests/examples/TypeSignatureParens.hs:3:81-91 }
               Just (Ann (DP (0,0)) (ColDelta 81) DP (0,0) [] [((G AnnVal),DP (0,0))])
               (HsTyVar 
                (Qual {ModuleName: LL} {OccName: ListLike}))) 
              ({ tests/examples/TypeSignatureParens.hs:3:93-97 }
               Just (Ann (DP (0,1)) (ColDelta 93) DP (0,1) [] [((G AnnVal),DP (0,0))])
               (HsTyVar 
                (Unqual {OccName: state}))))) 
            ({ tests/examples/TypeSignatureParens.hs:3:99 }
             Just (Ann (DP (0,1)) (ColDelta 99) DP (0,1) [] [((G AnnVal),DP (0,0))])
             (HsTyVar 
              (Unqual {OccName: a})))))]) 
        ({ tests/examples/TypeSignatureParens.hs:3:105-142 }
         Just (Ann (DP (0,1)) (ColDelta 105) DP (0,1) [] [((G AnnRarrow),DP (0,1))])
         (HsFunTy 
          ({ tests/examples/TypeSignatureParens.hs:3:105-107 }
           Just (Ann (DP (0,0)) (ColDelta 105) DP (0,0) [] [((G AnnOpenS),DP (0,0)),((G AnnCloseS),DP (0,0))])
           (HsListTy 
            ({ tests/examples/TypeSignatureParens.hs:3:106 }
             Just (Ann (DP (0,0)) (ColDelta 106) DP (0,0) [] [((G AnnVal),DP (0,0))])
             (HsTyVar 
              (Unqual {OccName: a}))))) 
          ({ tests/examples/TypeSignatureParens.hs:3:112-142 }
           Just (Ann (DP (0,1)) (ColDelta 112) DP (0,1) [] [((G AnnRarrow),DP (0,1))])
           (HsFunTy 
            ({ tests/examples/TypeSignatureParens.hs:3:112-114 }
             Just (Ann (DP (0,0)) (ColDelta 112) DP (0,0) [] [((G AnnVal),DP (0,0))])
             (HsTyVar 
              (Unqual {OccName: Int}))) 
            ({ tests/examples/TypeSignatureParens.hs:3:119-142 }
             Just (Ann (DP (0,1)) (ColDelta 119) DP (0,1) [] [])
             (HsAppTy 
              ({ tests/examples/TypeSignatureParens.hs:3:119-138 }
               Just (Ann (DP (0,0)) (ColDelta 119) DP (0,0) [] [])
               (HsAppTy 
                ({ tests/examples/TypeSignatureParens.hs:3:119 }
                 Just (Ann (DP (0,0)) (ColDelta 119) DP (0,0) [] [((G AnnVal),DP (0,0))])
                 (HsTyVar 
                  (Unqual {OccName: P}))) 
                ({ tests/examples/TypeSignatureParens.hs:3:121-138 }
                 Just (Ann (DP (0,1)) (ColDelta 121) DP (0,1) [] [((G AnnOpenP),DP (0,0)),((G AnnCloseP),DP (0,0))])
                 (HsParTy 
                  ({ tests/examples/TypeSignatureParens.hs:3:122-137 }
                   Just (Ann (DP (0,0)) (ColDelta 122) DP (0,0) [] [])
                   (HsAppTy 
                    ({ tests/examples/TypeSignatureParens.hs:3:122-133 }
                     Just (Ann (DP (0,0)) (ColDelta 122) DP (0,0) [] [])
                     (HsAppTy 
                      ({ tests/examples/TypeSignatureParens.hs:3:122-127 }
                       Just (Ann (DP (0,0)) (ColDelta 122) DP (0,0) [] [])
                       (HsAppTy 
                        ({ tests/examples/TypeSignatureParens.hs:3:122-124 }
                         Just (Ann (DP (0,0)) (ColDelta 122) DP (0,0) [] [((G AnnVal),DP (0,0))])
                         (HsTyVar 
                          (Unqual {OccName: Str}))) 
                        ({ tests/examples/TypeSignatureParens.hs:3:127 }
                         Just (Ann (DP (0,2)) (ColDelta 127) DP (0,2) [] [((G AnnVal),DP (0,0))])
                         (HsTyVar 
                          (Unqual {OccName: a}))))) 
                      ({ tests/examples/TypeSignatureParens.hs:3:129-133 }
                       Just (Ann (DP (0,1)) (ColDelta 129) DP (0,1) [] [((G AnnVal),DP (0,0))])
                       (HsTyVar 
                        (Unqual {OccName: state}))))) 
                    ({ tests/examples/TypeSignatureParens.hs:3:135-137 }
                     Just (Ann (DP (0,1)) (ColDelta 135) DP (0,1) [] [((G AnnVal),DP (0,0))])
                     (HsTyVar 
                      (Unqual {OccName: loc}))))))))) 
              ({ tests/examples/TypeSignatureParens.hs:3:140-142 }
               Just (Ann (DP (0,1)) (ColDelta 140) DP (0,1) [] [((G AnnOpenS),DP (0,0)),((G AnnCloseS),DP (0,0))])
               (HsListTy 
                ({ tests/examples/TypeSignatureParens.hs:3:141 }
                 Just (Ann (DP (0,0)) (ColDelta 141) DP (0,0) [] [((G AnnVal),DP (0,0))])
                 (HsTyVar 
                  (Unqual {OccName: a}))))))))))))) 
      (PlaceHolder)))),
   ({ tests/examples/TypeSignatureParens.hs:4:1-22 }
    Just (Ann (DP (1,1)) (ColDelta 1) DP (1,1) [] [])
    (ValD 
     (FunBind 
      ({ tests/examples/TypeSignatureParens.hs:4:1-10 }
       Just (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnVal),DP (0,0))])
       (Unqual {OccName: pTokenCost})) 
      (False) 
      (MG 
       [
        ({ tests/examples/TypeSignatureParens.hs:4:1-22 }
         Just (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnEqual),DP (0,1))])
         (Match 
          (Just 
           ((,) 
            ({ tests/examples/TypeSignatureParens.hs:4:1-10 }
             Just (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnVal),DP (0,0))])
             (Unqual {OccName: pTokenCost})) 
            (False))) 
          [
           ({ tests/examples/TypeSignatureParens.hs:4:12-13 }
            Just (Ann (DP (0,1)) (ColDelta 12) DP (0,1) [] [((G AnnVal),DP (0,0))])
            (VarPat 
             (Unqual {OccName: as}))),
           ({ tests/examples/TypeSignatureParens.hs:4:15-18 }
            Just (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])
            (VarPat 
             (Unqual {OccName: cost})))] 
          (Nothing) 
          (GRHSs 
           [
            ({ tests/examples/TypeSignatureParens.hs:4:20-22 }
             Just (Ann (DP (0,-1)) (ColDelta 20) DP (0,-1) [] [])
             (GRHS 
              [] 
              ({ tests/examples/TypeSignatureParens.hs:4:22 }
               Just (Ann (DP (0,1)) (ColDelta 22) DP (0,1) [] [((G AnnVal),DP (0,0))])
               (HsOverLit {HsOverLit:5}))))] 
           (EmptyLocalBinds))))] 
       [] 
       (PlaceHolder) 
       (FromSource)) 
      (WpHole) 
      (PlaceHolder) 
      [])))] 
  (Nothing) 
  (Nothing)))
==============
[(AnnKey tests/examples/TypeSignatureParens.hs:1:1 CN "HsModule",
  (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnEofPos),DP (1,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:1-10 CN "Unqual",
  (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:1-143 CN "SigD",
  (Ann (DP (2,1)) (ColDelta 1) DP (2,1) [DComment (DP (0,0),DP (0,36)) "{-# LANGUAGE ScopedTypeVariables #-}"] [((G AnnDcolon),DP (0,1))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:15-143 CN "HsForAllTy",
  (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnForall),DP (0,0)),((G AnnDot),DP (0,1))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:22-24 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 22) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:26-30 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 26) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:32 CN "UserTyVar",
  (Ann (DP (0,1)) (ColDelta 32) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:36-100 CN "(:)",
  (Ann (DP (0,1)) (ColDelta 36) DP (0,1) [] [((G AnnOpenP),DP (0,0)),((G AnnCloseP),DP (0,0)),((G AnnDarrow),DP (0,1))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:37-40 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 37) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:37-42 CN "HsAppTy",
  (Ann (DP (0,0)) (ColDelta 37) DP (0,0) [] [((G AnnComma),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:42 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 42) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:45-46 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 45) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:45-48 CN "HsAppTy",
  (Ann (DP (0,1)) (ColDelta 45) DP (0,1) [] [((G AnnComma),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:48 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 48) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:52-54 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 52) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:52-78 CN "HsOpTy",
  (Ann (DP (0,2)) (ColDelta 52) DP (0,2) [] [((G AnnComma),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:56-76 CN "Unqual",
  (Ann (DP (0,1)) (ColDelta 56) DP (0,1) [] [((G AnnBackquote),DP (0,0)),((G AnnVal),DP (0,0)),((G AnnBackquote),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:78 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 78) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:81-91 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 81) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:81-97 CN "HsAppTy",
  (Ann (DP (0,0)) (ColDelta 81) DP (0,0) [] [])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:81-99 CN "HsAppTy",
  (Ann (DP (0,1)) (ColDelta 81) DP (0,1) [] [])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:93-97 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 93) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:99 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 99) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:105-107 CN "HsListTy",
  (Ann (DP (0,0)) (ColDelta 105) DP (0,0) [] [((G AnnOpenS),DP (0,0)),((G AnnCloseS),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:105-142 CN "HsFunTy",
  (Ann (DP (0,1)) (ColDelta 105) DP (0,1) [] [((G AnnRarrow),DP (0,1))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:106 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 106) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:112-114 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 112) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:112-142 CN "HsFunTy",
  (Ann (DP (0,1)) (ColDelta 112) DP (0,1) [] [((G AnnRarrow),DP (0,1))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:119 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 119) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:119-138 CN "HsAppTy",
  (Ann (DP (0,0)) (ColDelta 119) DP (0,0) [] [])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:119-142 CN "HsAppTy",
  (Ann (DP (0,1)) (ColDelta 119) DP (0,1) [] [])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:121-138 CN "HsParTy",
  (Ann (DP (0,1)) (ColDelta 121) DP (0,1) [] [((G AnnOpenP),DP (0,0)),((G AnnCloseP),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:122-124 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 122) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:122-127 CN "HsAppTy",
  (Ann (DP (0,0)) (ColDelta 122) DP (0,0) [] [])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:122-133 CN "HsAppTy",
  (Ann (DP (0,0)) (ColDelta 122) DP (0,0) [] [])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:122-137 CN "HsAppTy",
  (Ann (DP (0,0)) (ColDelta 122) DP (0,0) [] [])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:127 CN "HsTyVar",
  (Ann (DP (0,2)) (ColDelta 127) DP (0,2) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:129-133 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 129) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:135-137 CN "HsTyVar",
  (Ann (DP (0,1)) (ColDelta 135) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:140-142 CN "HsListTy",
  (Ann (DP (0,1)) (ColDelta 140) DP (0,1) [] [((G AnnOpenS),DP (0,0)),((G AnnCloseS),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:3:141 CN "HsTyVar",
  (Ann (DP (0,0)) (ColDelta 141) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:4:1-10 CN "Unqual",
  (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:4:1-22 CN "Match",
  (Ann (DP (0,0)) (ColDelta 1) DP (0,0) [] [((G AnnEqual),DP (0,1))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:4:1-22 CN "ValD",
  (Ann (DP (1,1)) (ColDelta 1) DP (1,1) [] [])),
 (AnnKey tests/examples/TypeSignatureParens.hs:4:12-13 CN "VarPat",
  (Ann (DP (0,1)) (ColDelta 12) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:4:15-18 CN "VarPat",
  (Ann (DP (0,1)) (ColDelta 15) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey tests/examples/TypeSignatureParens.hs:4:20-22 CN "GRHS",
  (Ann (DP (0,-1)) (ColDelta 20) DP (0,-1) [] [])),
 (AnnKey tests/examples/TypeSignatureParens.hs:4:22 CN "HsOverLit",
  (Ann (DP (0,1)) (ColDelta 22) DP (0,1) [] [((G AnnVal),DP (0,0))])),
 (AnnKey <no location info> CN "EmptyLocalBinds",
  (Ann (DP (-4,0)) (ColDelta 0) DP (-4,0) [] []))]
==============
([((tests/examples/TypeSignatureParens.hs:3:1-143, AnnDcolon),
   [tests/examples/TypeSignatureParens.hs:3:12-13]),
  ((tests/examples/TypeSignatureParens.hs:3:1-143, AnnSemi),
   [tests/examples/TypeSignatureParens.hs:4:1]),
  ((tests/examples/TypeSignatureParens.hs:3:15-143, AnnDot),
   [tests/examples/TypeSignatureParens.hs:3:34]),
  ((tests/examples/TypeSignatureParens.hs:3:15-143, AnnForall),
   [tests/examples/TypeSignatureParens.hs:3:15-20]),
  ((tests/examples/TypeSignatureParens.hs:3:35-143, AnnCloseP),
   [tests/examples/TypeSignatureParens.hs:3:143]),
  ((tests/examples/TypeSignatureParens.hs:3:35-143, AnnOpenP),
   [tests/examples/TypeSignatureParens.hs:3:35]),
  ((tests/examples/TypeSignatureParens.hs:3:36-100, AnnCloseP),
   [tests/examples/TypeSignatureParens.hs:3:100]),
  ((tests/examples/TypeSignatureParens.hs:3:36-100, AnnDarrow),
   [tests/examples/TypeSignatureParens.hs:3:102-103]),
  ((tests/examples/TypeSignatureParens.hs:3:36-100, AnnOpenP),
   [tests/examples/TypeSignatureParens.hs:3:36]),
  ((tests/examples/TypeSignatureParens.hs:3:37-42, AnnComma),
   [tests/examples/TypeSignatureParens.hs:3:43]),
  ((tests/examples/TypeSignatureParens.hs:3:45-48, AnnComma),
   [tests/examples/TypeSignatureParens.hs:3:49]),
  ((tests/examples/TypeSignatureParens.hs:3:52-78, AnnComma),
   [tests/examples/TypeSignatureParens.hs:3:79]),
  ((tests/examples/TypeSignatureParens.hs:3:56-76, AnnBackquote),
   [tests/examples/TypeSignatureParens.hs:3:56,
    tests/examples/TypeSignatureParens.hs:3:76]),
  ((tests/examples/TypeSignatureParens.hs:3:56-76, AnnVal),
   [tests/examples/TypeSignatureParens.hs:3:57-75]),
  ((tests/examples/TypeSignatureParens.hs:3:105-107, AnnCloseS),
   [tests/examples/TypeSignatureParens.hs:3:107]),
  ((tests/examples/TypeSignatureParens.hs:3:105-107, AnnOpenS),
   [tests/examples/TypeSignatureParens.hs:3:105]),
  ((tests/examples/TypeSignatureParens.hs:3:105-142, AnnRarrow),
   [tests/examples/TypeSignatureParens.hs:3:109-110]),
  ((tests/examples/TypeSignatureParens.hs:3:112-142, AnnRarrow),
   [tests/examples/TypeSignatureParens.hs:3:116-117]),
  ((tests/examples/TypeSignatureParens.hs:3:121-138, AnnCloseP),
   [tests/examples/TypeSignatureParens.hs:3:138]),
  ((tests/examples/TypeSignatureParens.hs:3:121-138, AnnOpenP),
   [tests/examples/TypeSignatureParens.hs:3:121]),
  ((tests/examples/TypeSignatureParens.hs:3:140-142, AnnCloseS),
   [tests/examples/TypeSignatureParens.hs:3:142]),
  ((tests/examples/TypeSignatureParens.hs:3:140-142, AnnOpenS),
   [tests/examples/TypeSignatureParens.hs:3:140]),
  ((tests/examples/TypeSignatureParens.hs:4:1-22, AnnEqual),
   [tests/examples/TypeSignatureParens.hs:4:20]),
  ((tests/examples/TypeSignatureParens.hs:4:1-22, AnnFunId),
   [tests/examples/TypeSignatureParens.hs:4:1-10]),
  ((tests/examples/TypeSignatureParens.hs:4:1-22, AnnSemi),
   [tests/examples/TypeSignatureParens.hs:5:1]),
  ((<no location info>, AnnEofPos),
   [tests/examples/TypeSignatureParens.hs:5:1])],
 [(<no location info>,
   [AnnBlockComment "{-# LANGUAGE ScopedTypeVariables #-}"])])